File: chapter-4/recursive_max.py
def max(list):
    if len(list) == 0:
        return None
    if len(list) == 1:
        return list[0]
    else:
        max_num = max(list[1:])
        return list[1] if list[0] > max_num else max_num


print(max([1, 2, 3, 4]))

File: chapter-4/recursive_max.js
function max(arr) {
  if (arr.length === 0) return None;
  if (arr.length === 1) return arr[0];
  else {
    const max_num = max(arr.slice(1));
    if (arr[0] > max_num) return arr[0];
    else return max_num;
  }
}

console.log(max([1, 2, 3, 4, 5]));
